AppCenterData/AppCenterDataTests/MSDataTests.m

```diff
@@ -622,10 +628,257 @@ - (void)testListWhenOffline {
                                  }
                                }];
   XCTAssertNotNil(actualPaginatedDocuments);
-  XCTAssertNotNil(actualPaginatedDocuments.currentPage.error);
-  XCTAssertNotNil(actualPaginatedDocuments.currentPage.error);
-  XCTAssertEqual(actualPaginatedDocuments.currentPage.error.domain, kMSACDataErrorDomain);
-  XCTAssertEqual(actualPaginatedDocuments.currentPage.error.code, MSACDataErrorHTTPError);
+  XCTAssertNil(actualPaginatedDocuments.currentPage.error);
+  XCTAssertEqual([[actualPaginatedDocuments currentPage] items].count, 0);
+}
+
+- (void)testListReturnsFromLocalStorageWhenOffline {
+
+  // If
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@""];
+
+  // Simulate being offline.
+  MS_Reachability *reachabilityMock = OCMPartialMock([MS_Reachability reachabilityForInternetConnection]);
+  OCMStub([reachabilityMock currentReachabilityStatus]).andReturn(NotReachable);
+  self.sut.dataOperationProxy.reachability = reachabilityMock;
+
+  // Mock cached token result.
+  MSTokenResult *tokenResult = [self mockTokenFetchingWithError:nil];
+
+  // Mock local storage.
+  id<MSDocumentStore> localStorageMock = OCMProtocolMock(@protocol(MSDocumentStore));
+  self.sut.dataOperationProxy.documentStore = localStorageMock;
+  MSPaginatedDocuments *expectedDocuments = [MSPaginatedDocuments new];
+  OCMStub([localStorageMock listWithToken:tokenResult partition:OCMOCK_ANY documentType:OCMOCK_ANY baseOptions:OCMOCK_ANY])
+      .andReturn(expectedDocuments);
+
+  // When
+  [MSData listDocumentsWithType:[MSDictionaryDocument class]
+                      partition:kMSPartitionTest
+              completionHandler:^(MSPaginatedDocuments *_Nonnull documents) {
+                // Then
+                XCTAssertEqualObjects(expectedDocuments, documents);
+                [expectation fulfill];
+              }];
+
+  // Then
+  [self waitForExpectationsWithTimeout:3
```

Comment:
Most other tests use a timeout of 1.  We'd prefer to use lower values, have you tried this test with a value of 1?  Is it flaky?  How about on CI?

Reply:
Also consider this a blanket for other tests; we should use a value of 1 where possible